### PR TITLE
style: use env and gha if over bash

### PR DESCRIPTION
**Pull request by AlexanderBrevig**
# This is purely a cosmetic change

This PR enforces the `use env in steps` style.
Furthermore, I have extracted the bash if statements out to gha if, this way it is more clear that a skip happened in the logs.

I have taken a pass at renaming in a way that makes sense to me now, and hopefully still makes sense after we slightly change our underlying technologies (for instance from GCR to GAR).

Let's take the opportunity to have a discussion and come to an agreement.

## It kind of does look clean though?
image

### CODE REVIEW

File: .github/workflows/push.yml [hunk 1]
```diff
@@ -38,105 +38,103 @@ jobs:
...
+        if: env.GHA_DOCKER_PUSH_IMAGE_TAG  == 'image_tag'
         run: |
-          if [[ "${{ inputs.repository_type }}" = "GCR" ]]; then
-            echo "GHA_DOCKER_PUSH_META_IMAGES=eu.gcr.io/${{ env.GHA_DOCKER_PUSH_GCR_PROJECT_ID }}/${{ env.GHA_DOCKER_PUSH_IMAGE_ARTIFACT_NAME }}" >> $GITHUB_ENV
-          elif [[ "${{ inputs.repository_type }}" = "ACR" ]]; then
-            echo "GHA_DOCKER_PUSH_META_IMAGES=acrentur001.azurecr.io/${{ env.GHA_DOCKER_PUSH_IMAGE_ARTIFACT_NAME }}" >> $GITHUB_ENV
-          fi
+          SHA=${{ github.sha }}
+          SHORT_SHA=${SHA:0:7}
+          echo "GHA_DOCKER_PUSH_IMAGE_TAG=${{ env.GHA_DOCKER_PUSH_BRANCH_NAME }}.${{ env.GHA_DOCKER_PUSH_DATE }}-SHA$SHORT_SHA" >> $GITHUB_ENV
```
Comment (Glenn-Terjesen) [inline .github/workflows/push.yml:83]: ..-SHA$SHORT_SHA => -SHA${SHORT_SHA} ?

File: .github/workflows/push.yml [hunk 2]
```diff
@@ -38,105 +38,103 @@ jobs:
...
+      - id: set-registry-azure
+        shell: bash
+        if: env.GHA_DOCKER_PUSH_CLOUD_PROVIDER  == 'Azure'
+        run: |
+          echo "GHA_DOCKER_PUSH_DOCKER_REGISTRY=acrentur001.azurecr.io" >> $GITHUB_ENV
+      - id: set-registry-gcp
+        shell: bash
+        if: env.GHA_DOCKER_PUSH_CLOUD_PROVIDER  == 'GCP'
+        run: |
+          echo "GHA_DOCKER_PUSH_DOCKER_REGISTRY=eu.gcr.io/entur-system-1287" >> $GITHUB_ENV
```
Comment (Glenn-Terjesen) [inline .github/workflows/push.yml:93]: registry, org secrets? (todo?)

**Review by Glenn-Terjesen — APPROVED** [review]

File: .github/workflows/push.yml (hunk 2 above)
Comment (AlexanderBrevig) [reply .github/workflows/push.yml:93]: What do you think @Spikra @bvestli? I agree.
Then we could simply redirect this to whatever secret name we decide for ACR and GCR (and cleanly swap when we go GAR).

Comment (bvestli) [reply .github/workflows/push.yml:93]: Agree on that 👍

File: .github/workflows/push.yml [hunk 3]
```diff
@@ -3,31 +3,31 @@ name: Entur/Docker/Push
 on:
   workflow_call:
     inputs:
-      image_artifact_name:
+      image_name:
```
Comment (bvestli) [inline .github/workflows/push.yml:6]: Actually it is the name of the GitHub artifact, not the image in the artifact. But the description is clear on that, so this is ok for me.

**Review by bvestli — APPROVED** [review]

File: .github/workflows/push.yml (hunk 3 above)
Comment (AlexanderBrevig) [reply .github/workflows/push.yml:6]: Currently it does not seem to use the previously built image. 
I am OK to merge it like this, and add an issue. If we _require_ a build step (like we do now) then we should at least re-use the work done by it :) 

Maybe the meta step needs to be in the build? Not sure, I guess we should check the docs again - I seem to remember they had an example for splitting build and push?

Comment (AlexanderBrevig) [reply .github/workflows/push.yml:6]: It is _also_ the actual name of the image, so it serves double duty. I think it's clearer when we emphasize the aspect that our developers are used to think about; the actual image name 👍🏻

Comment (bvestli) [reply .github/workflows/push.yml:6]: The image in the GitHub artifact is 'image_name'.tar.

And it is used in the build step. Ref. https://docs.docker.com/build/ci/github-actions/test-before-push/

Comment (bvestli) [reply .github/workflows/push.yml:6]: If I understand the example correct

Comment (bvestli) [reply .github/workflows/push.yml:6]: But when looking at the job output I am not sure. https://github.com/entur/gha-docker/actions/runs/8442415988/job/23123788816#step:19:229
🤔

**Pull request merged**